serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc

```diff
@@ -0,0 +1,359 @@
+
+Dependency:
+.Callback state example with timeout
+[source, xml]
+----
+<dependency>
+    <groupId>org.kie.kogito</groupId>
+    <artifactId>kogito-addons-quarkus-jobs-knative-eventing</artifactId>
+</dependency>
+----
+
+Configuration parameters:
+
+.Callback state example with timeout
+[source, properties]
+----
+# Events produced by kogito-addons-quarkus-jobs-knative-eventing to program the timers on the job service.
+mp.messaging.outgoing.kogito-job-service-job-request-events.connector=quarkus-http
+mp.messaging.outgoing.kogito-job-service-job-request-events.url=${K_SINK:http://localhost:8280/jobs/events}
+mp.messaging.outgoing.kogito-job-service-job-request-events.method=POST
+----
+
+[#timeout-example]
+== Timeout showcase example
+
+In the link:{kogito_sw_examples_url}/serverless-workflow-timeouts-showcase[serverless-workflow-timeouts-showcase] you can see a complete architecture, containing a serverless workflow application with timeouts configured alongside Job service running on Knative.
+
+There are two workflows to showcase timeouts usage: `Callback` and `Switch` states.
+
+=== Callback workflow
+
+It is a simple workflow where once the execution reaches the callback state it waits for the event `callbackEvent`to be received and continue the execution.
+
+.Callback timeout workflow
+image::core/callback-state-timeouts.svg[]
+
+.Callback event
+[source, json]
+----
+{
+"name": "callbackEvent",
+"source": "",
+"type": "callback_event_type"
+}
+----
+
+A timeout is configured with a maximum time 30 seconds to be waited by the workflow to receive the callbackEvent, in case it does not arrive in time, the execution moves, and the eventData variable remains null.
+See the <<callback-state, callback state definition>>.
+
+
+=== Switch workflow
+
+The switch example is similar to the callback but once the execution reaches the state, it waits for once of the two configured events, `visaDeniedEvent` or `visaApprovedEvent`, to be consumed, see the <<switch-state, switch state definition>>.
+
+Whenever one of the events is consumed, before the configured timeout, the workflow execution moves to the next state defined in the `transition`.
+
+If none of the events arrive before 30 seconds of timeout, the workflow then moves to the state defined in `defaultCondition' transition.
+
+.Switch timeout workflow
+image::core/switch-state-timeouts.svg[]
+
+=== Running the example
+
+In order to run the example you need to have a kubernetes or Openshift cluster running with Knative configured. In the example minikube is being used you can follow the steps described in the example's link:{kogito_sw_examples_url}/serverless-workflow-timeouts-showcase[readme].
+
+All the descriptor files used to deploy the example infrastructure are present in the kubernetes folder, and for the workflow application, the descriptors are placed on the target/kubernetes, generated after the build.
```

Comment:
Maybe here we can add a [NOTE] ?
And spit a bit phrase, like "The descriptor related to the showcase-databse and jobs service deployment are located at....
The descriptors related to the application....